frontend/src/metabase/query_builder/actions/downloading.ts

```diff
@@ -144,12 +145,15 @@ const getDatasetParams = ({
 
 export function getDatasetDownloadUrl(url: string, params?: URLSearchParams) {
   url = url.replace(api.basename, ""); // make url relative if it's not
-  url = api.basename + url;
   if (params) {
     url += `?${params.toString()}`;
   }
-  const requestUrl = new URL(url, location.origin);
-  return requestUrl.href;
+
+  return url;
```

Comment:
We need to ensure we now only return the relative path without the subpath. Previously we needed the whole path because we passed it directly to `fetch` but this PR passed it to `GET` and `POST` which already take care of the subpath via api.basename.

i.e. if we deploy metabase on `/metabase` and the download URl is `/metabase/embed/question/123.csv`. This function needs to return just `/embed/question/123.csv` as you'll see in the [unit test below](https://github.com/metabase/metabase/pull/41633/files#diff-35fae1f6b35bce152740771ce85acc17c4670aa93f88c3bbee9ddc222c059b20R48)